app/src/sharedTest/java/org/oppia/android/app/testing/SplashTestActivityTest.kt

```diff
@@ -0,0 +1,202 @@
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG
+import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_VALUE
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [SplashTestActivity]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = SplashTestActivityTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class SplashTestActivityTest {
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var platformParameterController: PlatformParameterController
+
+  private val mockPlatformParameterList by lazy {
+    val mockSplashScreenWelcomeMsgParam = PlatformParameter.newBuilder()
+      .setName(SPLASH_SCREEN_WELCOME_MSG)
+      .setBoolean(SPLASH_SCREEN_WELCOME_MSG_VALUE)
+      .build()
+
+    listOf<PlatformParameter>(
+      mockSplashScreenWelcomeMsgParam
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+  }
+
+  @Test
+  fun testSplashTestActivity_readEmptyDatabase_checkWelcomeMsgIsInvisibleByDefault() {
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG)).check(doesNotExist())
+    }
+  }
+
+  @Test
+  fun testSplashTestActivity_updateEmptyDatabase_readDatabaseValues_checkWelcomeMsgIsVisible() {
+    platformParameterController.updatePlatformParameterDatabase(mockPlatformParameterList)
+    testCoroutineDispatchers.runCurrent()
+
+    launch(SplashTestActivity::class.java).use {
+      onView(withText(SplashTestActivity.WELCOME_MSG))
+        .inRoot(getToastMatcher())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  private fun getToastMatcher(): TypeSafeMatcher<Root?> {
+    return object : TypeSafeMatcher<Root?>() {
+      override fun describeTo(description: Description?) {
+        description?.appendText("is toast")
+      }
+
+      override fun matchesSafely(item: Root?): Boolean {
+        val type: Int? = item?.windowLayoutParams?.get()?.type
+        if (type == WindowManager.LayoutParams.TYPE_TOAST) {
+          val windowToken: IBinder = item.decorView.windowToken
+          val appToken: IBinder = item.decorView.applicationWindowToken
+          if (windowToken === appToken) { // means this window isn't contained by any other windows.
```

Comment:
```
if (windowToken === appToken) {
  // This window isn't contained by any other windows.
```

Reply:
Done

Reply:
Please write the comment as a sentence`// Means this window isn't contained by any other windows.`

Reply:
Done